daemon/container.go

```diff
@@ -1034,28 +1034,23 @@ func (container *Container) Copy(resource string) (io.ReadCloser, error) {
 		}
 	}()
 
+	if len(resource) == 0 { // Copying `/` we expand to copy its content
```

Comment:
this actually is the case when you do `docker cp ID:/ /tmp`, copying the whole file system. We strip the first `/` at the api level and arrive to this point with an empty string. Docker already fails if you try to copy an empty string, but I've added a test case to be explicit about it.